plugins/account_history/plugin.cpp

```diff
@@ -135,242 +157,290 @@ if( options.count(name) ) { \
-            impacted.insert(op.from);
+            impacted.insert(make_pair(op.from, operation_direction_type::any));
         }
 
         void operator()(const decline_voting_rights_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::any));
         }
 
         void operator()(const comment_benefactor_reward_operation& op) {
-            impacted.insert(op.benefactor);
-            impacted.insert(op.author);
+            impacted.insert(make_pair(op.benefactor, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.author, operation_direction_type::sender));
         }
 
         void operator()(const delegate_vesting_shares_operation& op) {
-            impacted.insert(op.delegator);
-            impacted.insert(op.delegatee);
+            impacted.insert(make_pair(op.delegator, operation_direction_type::sender));
+            impacted.insert(make_pair(op.delegatee, operation_direction_type::receiver));
         }
 
         void operator()(const return_vesting_delegation_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::receiver));
         }
 
         void operator()(const proposal_create_operation& op) {
-            impacted.insert(op.author);
+            impacted.insert(make_pair(op.author, operation_direction_type::any));
         }
 
         void operator()(const proposal_update_operation& op) {
-            impacted.insert(op.active_approvals_to_add.begin(), op.active_approvals_to_add.end());
-            impacted.insert(op.owner_approvals_to_add.begin(), op.owner_approvals_to_add.end());
-            impacted.insert(op.posting_approvals_to_add.begin(), op.posting_approvals_to_add.end());
-            impacted.insert(op.active_approvals_to_remove.begin(), op.active_approvals_to_remove.end());
-            impacted.insert(op.owner_approvals_to_remove.begin(), op.owner_approvals_to_remove.end());
-            impacted.insert(op.posting_approvals_to_remove.begin(), op.posting_approvals_to_remove.end());
+            auto insert_fn = [this](const fc::flat_set<golos::chain::account_name_type>& impd) {
+                for (auto i : impd) {
+                    impacted.insert(make_pair(i, operation_direction_type::any));
+                }
+            };
+            insert_fn(op.active_approvals_to_add);
+            insert_fn(op.owner_approvals_to_add);
+            insert_fn(op.posting_approvals_to_add);
+            insert_fn(op.active_approvals_to_remove);
+            insert_fn(op.owner_approvals_to_remove);
+            insert_fn(op.posting_approvals_to_remove);
         }
 
         void operator()(const proposal_delete_operation& op) {
-            impacted.insert(op.requester);
+            impacted.insert(make_pair(op.requester, operation_direction_type::any));
```

Comment:
here should be `sender` too.
proposal author can be `receiver` if it differs from `op.requester`